crates/core/src/ring.rs

```diff
@@ -417,17 +438,14 @@ impl Ring {
         } else if open_conn >= self.max_connections {
             false
         } else {
-            let strategy = if self
-                .fast_acquisition
-                .load(std::sync::atomic::Ordering::Acquire)
-            {
-                AcquisitionStrategy::Fast
-            } else {
-                AcquisitionStrategy::Slow
-            };
+            let strategy = self
+                .topology_manager
+                .read()
+                .get_connection_acquisition_strategy();
+
             self.topology_manager
                 .write()
-                .evaluate_new_connection(location, strategy)
+                .evaluate_new_connection(location, strategy, Instant::now())
```

Comment:
why separate this is 2 different function calls? If topolofy_manager already has access to the strategy itself isn't a bit redundant explicitly gettign the strategy and passing it?

Reply:
Yes, my mistake - fixed.